dev/src/codewind/connection/Connection.ts

```diff
@@ -52,10 +54,16 @@ export default class Connection implements vscode.QuickPickItem, vscode.Disposab
         this.workspacePath = vscode.Uri.file(cwEnv.workspace);
         this.versionStr = CWEnvironment.getVersionAsString(cwEnv.version);
         this.host = this.getHost(url);
+        this.remote = true;
 
         // caller must await on this promise before expecting this connection to function correctly
         // it does happen very quickly (< 1s) but be aware of potential race here
-        this.initFileWatcherPromise = this.initFileWatcher();
+        if (!this.remote) {
+            this.initFileWatcherPromise = this.initFileWatcher();
+        } else {
+            // Disable file-watcher in remote mode for now.
+            this.initFileWatcherPromise = new Promise<void>((resolve) => (resolve()));
```

Comment:
Minor but you can just use 

` this.initFileWatcherPromise = Promise.resolve();`